modules/experimental-layers/src/contour-layer/marching-squares.js

```diff
@@ -32,57 +32,66 @@ const CODE_OFFSET_MAP = {
 };
 
 // Returns marching square code for given cell
+/* eslint-disable complexity */
 export function getCode(params) {
   // Assumptions
   // Origin is on bottom-left , and X increase to right, Y to top
   // When processing one cell, we process 4 cells, by extending row to top and on column to right
   // to create a 2X2 cell grid
 
-  const {cellWeights, thresholdValue, cellIndex, gridSize} = params;
-
-  const numRows = gridSize[1];
-  const numCols = gridSize[0];
-
-  // TODO: duplicate top row and right column
-  // We shouldn't process the right column
-  assert((cellIndex + 1) % numCols);
-  // We shouldn't process the topmost row
-  assert(cellIndex + 1 < (numRows - 1) * numCols);
-
-  const top = cellWeights[cellIndex + numCols] - thresholdValue >= 0 ? 1 : 0;
-  const topRight = cellWeights[cellIndex + numCols + 1] - thresholdValue >= 0 ? 1 : 0;
-  const right = cellWeights[cellIndex + 1] - thresholdValue >= 0 ? 1 : 0;
-  const current = cellWeights[cellIndex] - thresholdValue >= 0 ? 1 : 0;
+  const {cellWeights, thresholdValue, x, y, width, height} = params;
+
+  assert(x >= -1 && x < width);
+  assert(y >= -1 && y < height);
+
+  const leftBoundary = x < 0;
```

Comment:
Nit: Maybe name these `isLeftBoundary...` to emphasize in the code below that they are booleans.